File: data/src/main/java/com/thinkzi/weather/data/exception/RepositoryErrorBundle.java
package com.thinkzi.weather.data.exception;

import com.thinkzi.weather.domain.exception.ErrorBundle;

/**
 * provide Wrapper around Exceptions used to manage errors in the repository.
 */
public class RepositoryErrorBundle implements ErrorBundle {

    private final Exception _exception;

    public RepositoryErrorBundle(Exception _exception) {
        this._exception = _exception;
    }

    @Override
    public Exception getException() {
        return _exception;
    }

    @Override
    public String getErrorMessage() {
        String _message = "";

        if (this._exception != null) {
            _message = this._exception.getMessage();
        }

        return _message;
    }

}

File: data/src/main/java/com/thinkzi/weather/data/exception/RepositoryErrorBundle.kt
package com.thinkzi.weather.data.exception

import com.thinkzi.weather.domain.exception.ErrorBundle

/**
 * provide Wrapper around Exceptions used to manage errors in the repository.
 */
class RepositoryErrorBundle(override val exception: Exception) : ErrorBundle {

    override val errorMessage: String
        get() {
            var _message: String = this.exception.message!!

            return _message
        }

}
